### PR TITLE
ddl: add some comments to help readability

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  most are comments, other are variable rename
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/model/ddl.go [hunk 1]
```diff
@@ -488,9 +488,13 @@ type JobMeta struct {
 
 // Job is for a DDL operation.
 type Job struct {
-	ID         int64         `json:"id"`
-	Type       ActionType    `json:"type"`
-	SchemaID   int64         `json:"schema_id"`
+	ID   int64      `json:"id"`
+	Type ActionType `json:"type"`
+	// SchemaID means different for different job types:
+	// - ExchangeTablePartition: nt db id
```
Comment (wjhuang2016) [inline pkg/parser/model/ddl.go:494]: what's 'nt'?

Comment (D3Hunter) [reply pkg/parser/model/ddl.go:494]: see https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html

File: pkg/meta/meta.go [hunk 2]
```diff
@@ -473,6 +474,8 @@ func (m *Meta) GetAutoIDAccessors(dbID, tableID int64) AutoIDAccessors {
 // To solve this problem, we always check the schema diff at first, if the diff is empty, we know at t2 moment we can only see the v9 schema,
 // so make neededSchemaVersion = neededSchemaVersion - 1.
 // For `Reload`, we can also do this: if the newest version's diff is not set yet, it is ok to load the previous version's infoSchema, and wait for the next reload.
+// if there are multiple consecutive jobs failed or cancelled after the schema version
```
[the remark this answers is not in the document]
Comment (lance6716) [reply pkg/meta/meta.go:477]: Do you think the name "GetSchemaVersionSkipPendingDiff" is better?

Comment (D3Hunter) [reply pkg/meta/meta.go:477]: a reverse of `WithNonEmptyDiff`, seems no difference

Comment (lance6716) [reply pkg/meta/meta.go:477]: as your comment said
> the returned 'version - 1' might still not have diff.

So this function will not always satisfy "WithNonEmptyDiff". And I read the comment it just want to skip one version that not set diff yet. That's why I think "SkipPendingDiff" is better

**Review by lance6716 — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**